packages/wallets/fractal/src/adapter.ts

```diff
@@ -0,0 +1,207 @@
+            this._connecting = false;
+        };
+
+        this.popupManager.onConnectionUpdated((connection) => {
+            if (!connection) {
+                return;
+            }
+            connection.on(PopupEvent.SOLANA_WALLET_ADAPTER_APPROVED, handleSolanaWalletAdapterApproved);
+        });
+
+        return new Promise((promiseResolver, promiseRejector) => {
+            resolve = promiseResolver;
+            reject = promiseRejector;
+        });
+    }
+
+    async disconnect(): Promise<void> {
+        this.popupManager.tearDown();
+        this._publicKey = null;
+        this.emit('disconnect');
+    }
+
+    async signTransaction<T extends Transaction>(transaction: T): Promise<T> {
+        try {
+            this.checkWalletReadiness();
+            const result = await this.signTransactions([transaction]);
+            return result[0];
+        } catch (error: any) {
+            let errorToThrow = error;
+            if (!(error instanceof WalletError)) {
+                errorToThrow = new WalletSignTransactionError(error?.message, error);
+            }
+            this.emit('error', errorToThrow);
+            throw error;
+        }
+    }
+
+    async signAllTransactions<T extends Transaction>(transactions: T[]): Promise<T[]> {
+        try {
+            this.checkWalletReadiness();
+            const result = await this.signTransactions(transactions);
+            return result;
+        } catch (error: any) {
+            let errorToThrow = error;
+            if (!(error instanceof WalletError)) {
+                errorToThrow = new WalletSignTransactionError(error?.message, error);
+            }
+            this.emit('error', errorToThrow);
+            throw error;
+        }
+    }
+
+    private async signTransactions<T extends Transaction>(transactions: T[]): Promise<T[]> {
+        let resolve: (signedTransactions: T[]) => void;
+        let reject: (err: WalletError) => void;
+
+        const handleTransactionSignatureNeededResponse = (payload: unknown) => {
+            if (!assertPayloadIsTransactionSignatureNeededResponsePayload(payload)) {
+                const error = new WalletSignTransactionError(
+                    'Malformed payload when signing transactions. ' +
+                        'Expected { signedB58Transactions: string[] } ' +
+                        `but received ${payload}`
+                );
+                reject(error);
+                this.emit('error', error);
+                return;
+            }
+
+            const signedTransactions = payload.signedB58Transactions.map((signedB58Transaction) => {
+                const message = Message.from(base58.decode(signedB58Transaction));
+                return Transaction.populate(message);
+            }) as T[];
+
+            resolve(signedTransactions);
+        };
+
+        const nonce = uuid.v4();
+        this.popupManager.open({
+            url: `${FRACTAL_DOMAIN_HTTPS}/wallet-adapter/sign/${nonce}`,
```

Comment:
should we make the whole URL minus the nonce a const?

Reply:
Done.